### PR TITLE
Medical Items and Services Admin pages

### PR DESCRIPTION
**Pull request by edarchis**

### CODE REVIEW

**Comment by qgerome** [comment]
Do you think we could format the date in the `valid from` & `valid to` column ?
image

**Comment by qgerome** [comment]
When I filter on service_type=curative & I order on valid_from I have this error:
image

**Comment by qgerome** [comment]
I tried to edit a service and I have this error:

**Comment by qgerome** [comment]
There is something fishy with this select. The default value is not one for the choices : 
image

**Comment by qgerome** [comment]
Shouldn't it be blank (or "All") instead of "None" ? IMO the meaning is not the same
image

**Comment by qgerome** [comment]
image
* Service level is the DB value and not the label to display to the user
* I'd put the same label for the frequency ("Frequency (days)") since it's hard to know the unit of time

**Comment by qgerome** [comment]
I don't know if it's due to a limitation but it's weird to be able to select "None" despite the fact that the input is required, no ? 
image

Same goes for this input as well: 
image
You can select "Any" but it means `null` and prevent the user to save the form.

File: src/actions.js [hunk 1]
```diff
@@ -36,4 +44,302 @@ export function fetchServicePicker(mm, str, refDate) {
...
+    ${ms.frequency ? `frequency: "${formatGQLString(ms.frequency)}"` : ""}
+    ${ms.patientCategory ? `patientCategory: ${ms.patientCategory}` : ""}
+    ${ms.category && ms.category !== " " ? `category: "${formatGQLString(ms.category)}"` : ""}
+    ${ms.level ? `level: "${formatGQLString(ms.level)}"` : ""}
+    ${ms.package ? `package: "${formatGQLString(ms.package)}"` : ""}
+  `;
+  return req;
+}
+
+export function fetchMedicalServices(mm, hf, str, prev) {
```
Comment (qgerome) [inline src/actions.js:109]: I was not able to find the place where it's used. How is it used ?

**Comment by qgerome** [comment]
Code looks ok to me.

**Comment by edarchis** [comment]
I agree with the Any vs None. I think that it's managed somehow by the components that we have but I'm not sure exactly how.

**Comment by edarchis** [comment]
>     * I'd put the same label for the frequency ("Frequency (days)") since it's hard to know the unit of time

See, I thought that it was in times/year but you're right, it's in days between occurrences... 👍

**Comment by edarchis** [comment]
Any is achieved by withNull=true and nullLabel="medical.xxx.any".
The withNull on mandatory fields allow them to be blank when creating a new record. Otherwise, we would need to specify the default value.

**Review by qgerome — APPROVED** [review]

**Pull request merged**